### ISSUE TITLE
[Windows 11 Taskbar Styler] Actually v1.3.4 fix made the bug worse

### ISSUE THREAD

**Issue by MuscularPuky** [ramensoftware/windhawk-mods#742]
sry for too late the issue but I was going to busy until Sep

in v1.3.3, there's acrylic background styling bug with multi monitor. v1.3.4 is the hoxfix for that multi monitor bug but it made the bug worse.

v1.3.3, only secondary taskbar lost background but now v1.3.4, all taskbars lost background. when I reload mod, the taskbars have default background in initial 0.1~0.2s but it gradually faded soon.

on single monitor, the acrylic background works well.

**Comment by m417z** [comment]
For some reason, setting an acrylic background for both taskbars doesn't work. Setting only of them, either one, works.

Here you can see setting both with a delay:

https://github.com/ramensoftware/windhawk-mods/assets/4129781/ef17eef0-233f-489f-9346-c97be409d713

**Comment by MuscularPuky** [comment]
> For some reason, setting an acrylic background for both taskbars doesn't work. Setting only of them, either one, works.
> 
> Here you can see setting both with a delay:
> 
>  ZFfxKfNQOO.mp4

THX for your hint I did it [#29](https://github.com/ramensoftware/windows-11-start-menu-styling-guide/issues/29)
it's not code issue. maybe kind of XAML limitation. but I don't know why it cause
anyways we can bypass this problem through add more parameter, and still solvable—without bypass—by edit the mod code—to use other hook method
very teresting parameters 🤔 I'll studying more about these

**Comment by m417z** [comment]
The following seems to work (suggested by @ahmed605):
* Add `BackgroundSource="HostBackdrop"` to the style, for example, target `Rectangle#BackgroundFill` and style:
```
Fill:=<AcrylicBrush BackgroundSource="HostBackdrop" TintColor="Transparent" TintOpacity="0" TintLuminosityOpacity="0" Opacity="1"/>
```
* Enable Host Backdrop for the taskbars as following:
```cpp
ACCENTPOLICY policy = { ACCENT_ENABLE_BLURBEHIND, 0, 0, 0 };
WINCOMPATTRDATA data = { WCA_ACCENT_POLICY, &policy, sizeof(policy) };
SetWindowCompositionAttribute(hWnd, &data);
```

I can make the mod to enable Host Backdrop for the taskbar, but I'm not sure what the implications are, especially since most users probably don't use a custom acrylic background.

**Comment by m417z** [comment]
My previous message is incorrect. I was about to try `ACCENT_ENABLE_HOSTBACKDROP`, but tried `ACCENT_ENABLE_BLURBEHIND` instead. `ACCENT_ENABLE_HOSTBACKDROP` doesn't seem to work.

What works is just setting `ACCENT_ENABLE_BLURBEHIND` with `SetWindowCompositionAttribute` (`BackgroundSource="HostBackdrop"` isn't necessary), but it also results in a slightly different effect.

**Comment by e391hed12** [comment]
Hi I'm having the exact same problem - making the background acrylic doesn't work, only transparent, default, or a solid color. I have two monitors (though it doesn't seem to work on a single one either).

> My previous message is incorrect. I was about to try `ACCENT_ENABLE_HOSTBACKDROP`, but tried `ACCENT_ENABLE_BLURBEHIND` instead. `ACCENT_ENABLE_HOSTBACKDROP` doesn't seem to work.
> 
> What works is just setting `ACCENT_ENABLE_BLURBEHIND` with `SetWindowCompositionAttribute` (`BackgroundSource="HostBackdrop"` isn't necessary), but it also results in a slightly different effect.

I don't have programming knowledge, could you perhaps tell me how to implement that? 

The only reason I'm using windhawk was to make the taskbar rounded and blurry, it'd be sucky for me if it only could do the first

**Comment by MuscularPuky** [comment]
> Hi I'm having the exact same problem - making the background acrylic doesn't work, only transparent, default, or a solid color. I have two monitors (though it doesn't seem to work on a single one either).
> 
> > My previous message is incorrect. I was about to try `ACCENT_ENABLE_HOSTBACKDROP`, but tried `ACCENT_ENABLE_BLURBEHIND` instead. `ACCENT_ENABLE_HOSTBACKDROP` doesn't seem to work.
> > What works is just setting `ACCENT_ENABLE_BLURBEHIND` with `SetWindowCompositionAttribute` (`BackgroundSource="HostBackdrop"` isn't necessary), but it also results in a slightly different effect.
> 
> I don't have programming knowledge, could you perhaps tell me how to implement that?
> 
> The only reason I'm using windhawk was to make the taskbar rounded and blurry, it'd be sucky for me if it only could do the first

it's known issue and unfortunately there's no silver bullet so far as I kwno

**Comment by m417z** [comment]
I've implemented the `ACCENT_ENABLE_BLURBEHIND` alternative in the following mod:

https://windhawk.net/mods/taskbar-background-helper

You can set the taskbar background to transparent and use it for handling the background instead.

**Comment by m417z** [comment]
I've implemented the `ACCENT_ENABLE_BLURBEHIND` alternative in the following mod: [Taskbar Background Helper](https://windhawk.net/mods/taskbar-background-helper).
You can set the taskbar background to transparent and use it for handling the background instead.

**Comment by JotaRata** [comment]
Taskbar Background Helper doesn't seem to work:(

**Comment by m417z** [comment]
@JotaRata what do you mean by that? What do you see? Did you try changing the settings?

**Comment by JotaRata** [comment]
The same I had with Taskbar Styler before. 
To test it, I temporarely set the theme to "Translucent taskbar" and enabled Taskbar Background Helper and I see the same semi-transparent solid color taskbar when using two monitors when using the Blur and Acrylic blur background styles.

![imagen](https://github.com/user-attachments/assets/ee617332-881c-4024-af28-da3edf01e683)

**Comment by m417z** [comment]
Did you maximize a window? By default the mod only sets a background when a window is maximized.

**Comment by JotaRata** [comment]
No, I disabled the option to only be active when having a maximized window

**Comment by JotaRata** [comment]
Setting the whole background to Transparent has the same effect:
(This is my second monitor, though the first looks the same)

![imagen](https://github.com/user-attachments/assets/f1e05cc4-ca6c-4864-af98-9c5ecb282734)

**Comment by m417z** [comment]
What about other background styles? Does anything else work?
Also, do you have "Transparency effects" enabled in system settings?

**Comment by JotaRata** [comment]
I have them enabled.
Other styles don't work neither, the only difference I see is when choosing a Solid color in Background Helper settings, I see that color when there were transparent areas before. Acrylic and Blur don't work

**Comment by m417z** [comment]
OK, so at least something works, the mod is loaded and isn't completely broken. No idea honestly, the mod just uses the standard Windows methods to set these styles. My next guess is that you're either running in some kind of an efficiency/battery saver mode, or that your graphic card is too weak and so Windows doesn't enable some styles.

**Comment by JotaRata** [comment]
I don't think that's the case, since transparency effects are indeed enabled on my system and translucent/blurry elements such as notifications and start menu are visible. Windhawk also corrrectly sets the translucent effects for the taskbar when using only one monitor.

The problem seems to arise when using two monitors as described previously in other GH issues.

**Comment by m417z** [comment]
> I don't think that's the case, since transparency effects are indeed enabled on my system and translucent/blurry elements such as notifications and start menu are visible. Windhawk also corrrectly sets the translucent effects for the taskbar when using only one monitor.

That transparency works differently on the OS level, so it's still possible that for some reason some environment condition on your computer doesn't affect that, but affects the method that the Taskbar Background Helper mod uses.

> The problem seems to arise when using two monitors as described previously in other GH issues

Yes, that's what this issue is about and it seems to happen for everyone. To sum it up in a table:

| → Setup<br>↓User   | Styler mod blur<br> Single monitor | Styler mod blur<br> Multiple monitors | Taskbar Background Helper<br>Single/multiple monitors |
|--------------------|------------------------------------|---------------------------------------|-------------------------------------------------------|
| Me and other users | ✅                                  | ❌                                     | ✅                                                     |
| You                | ✅                                  | ❌                                     | ❌                                                     |

**Comment by romariossilva** [comment]
turn off the settings "show my taskbar on all display" and it will work.

**Comment by highesttt** [comment]
> turn off the settings "show my taskbar on all display" and it will work.

That's typically a setting we don't want to turn off...

**Comment by nekusu** [comment]
Would it be feasible to add an option to select specific monitors we want to apply the styles to? That way we could set the acrylic background only on the main taskbar and a solid color for the secondary taskbars. This could be a workaround for this bug and also expand the styling possibilities with the mod.

**Comment by m417z** [comment]
Windows 11 Taskbar Styler v1.4 now adds a custom blur brush implementation, which can be used as following:
```
Fill:=<WindhawkBlur BlurAmount="10" TintColor="#80FF00FF"/>
```

It doesn't have the limitations of the native `AcrylicBrush` brush, and can be used as a workaround.
